src/main/java/org/reactome/addlinks/fileprocessors/ensembl/EnsemblBioMartFileProcessor.java

```diff
@@ -0,0 +1,167 @@
+    @Override
+    public Map<String, Map<String, List<String>>> getIdMappingsFromFile() {
+
+        // Below is a sample of the structure of the final mappings object. Each species has up to 4 different mappings associated with it.
+        /*
+             +hsapiens_uniprotToProteins
+             |              |
+             |              +---Q123456
+             |              |       |
+             |              |       + ENSP00001, ENSP00002,...
+             |              |
+             |              +---Q789012
+             |                      |
+             |                      + ENSP00003, ENSP00004,...
+             |
+             +ggallus_proteinsToTranscripts
+             |              |
+             |              +---ENSGALP00017
+             |              |       |
+             |              |       + ENSGALT00005, ENSGALT00006,...
+             |              |
+             |              +---ENSGALP00018
+             |                      |
+             |                      + ENSGALT00007, ENSGALT00008,...
+             +hsapiens_transcriptsToMicroarrays
+             |              |
+             |              +---ENST000019
+             |              |       |
+             |              |       + ILMN_123456, ILMN_7891011,...
+             |              |
+             |              +---ENST000020
+             |                      |
+             |                      + ILMN_121314, ILMN_141617,...
+             |
+             +mmusculus_proteinsToGenes
+             |              |
+             |              +---ENSMUSP000020
+             |              |       |
+             |              |       + ENSMUSG00009, ENSMUSG00010,...
+             |              |
+             |              +---ENSMUSP000021
+             |                      |
+             |                      + ENSMUSG00011, ENSMUSG00012,...
+
+         */
+
+        Map<String, Map<String, List<String>>> mappings = new HashMap<>();
+        // Read each file in BioMart directory and iterate through each line.
+        try (Stream<Path> paths = Files.walk(this.pathToFile)) {
+            paths
+                .filter(Files::isRegularFile)
+                .forEach(bioMartFile -> {
+                    // Get species name from filename
+                    String biomartFileSpecies = bioMartFile.getFileName().toString().split("_")[0];
+                    // Mappings that will be generated from each file for the species.
+                    Map<String, List<String>> proteinToTranscripts = new HashMap<>();
+                    Map<String, List<String>> proteinToGenes = new HashMap<>();
+                    Map<String, List<String>> transcriptToMicroarrays = new HashMap<>();
+                    Map<String, List<String>> uniprotToProteins = new HashMap<>();
+                    try {
+                        for (String biomartLine : Files.readAllLines(bioMartFile)) {
```

Comment:
A helper method to get the lines from `bioMartFile` would be useful to contain the handling and reduce the scope of the try block:

```
private List<String> getBioMartFileLines(Path bioMartFile) {
    List<String> bioMartFileLines;
    try {
        bioMartFileLines = Files.readAllLines(bioMartFile);
    } catch (IOException e) {
        logger.error("Unable to read file lines from " + getBioMartFile().toAbsolutePath().toString(), e);
        bioMartFileLines = new ArrayList<>();
    }
    return bioMartFileLines;
}

Reply:
I just used the existing method in BioMartUtils